src/sentry/grouping/api.py

```diff
@@ -100,15 +101,27 @@ def _get_enhancements(self, project: Project) -> str:
         cache_prefix = self.cache_prefix
         cache_prefix += f"{LATEST_VERSION}:"
         cache_key = (
-            cache_prefix + md5_text(f"{enhancements_base}|{project_enhancements}").hexdigest()
+            cache_prefix
+            + md5_text(
+                f"{enhancements_base}|{automatic_enhancements}|{project_enhancements}"
+            ).hexdigest()
         )
         enhancements = cache.get(cache_key)
         if enhancements is not None:
             return enhancements
 
         try:
+            # Automatic enhancements are always applied first, so they take precedence over
+            # project-specific enhancements.
+            enhancements_string = project_enhancements or ""
+            if automatic_enhancements:
+                enhancements_string = (
+                    f"{automatic_enhancements}\n{enhancements_string}"
+                    if enhancements_string
+                    else automatic_enhancements
+                )
```

Comment:
Nit: This can be simpler. If `enhancements_string` is falsy, the worst that happens is we add an extra newline to the auto enhancements.

```suggestion
                enhancements_string = f"{automatic_enhancements}\n{enhancements_string}"
```